test/TEST-16-DMSQUASH/test.sh

```diff
@@ -19,10 +20,16 @@ test_run() {
     "$testdir"/run-qemu \
         "${disk_args[@]}" \
         -boot order=d \
-        -append "rd.live.image rd.live.overlay.overlayfs=1 rd.live.dir=testdir root=LABEL=dracut console=ttyS0,115200n81 quiet selinux=0 rd.info rd.shell=0 panic=1 oops=panic softlockup_panic=1 $DEBUGFAIL" \
```

Comment:
Thanks for pointing that out. I originally had created a separate `TEST-18-DMSQUASH-AUTOOVERLAY` by duplicating the entirety of this test and adding my (minor) changes. I didn't feel good about all that copypasta, so I switched to this approach.

I hadn't checked out the NFS test and that's definitely a better way to go.

I'll adjust this test to use a similar approach.

Reply:
Thank you for considering it. 
https://github.com/dracutdevs/dracut/commit/027eee902d0d7af4db45fcb7835d49990c9fafa1 just landed, which hopefully helps you here as well.

Reply:
I've updated the test to add a separate QEMU run that uses its own initrd with my new module and modified the `init` script to only write the autooverlay marker if it's booted with the related kernel parameter. That way, the original test is unaltered.

I followed the example of the commit you linked to that updated this test script instead of using separate functions like `TEST-20-NFS`.